### PR TITLE
feat: add batch-prover Dockerfile and test

### PR DESCRIPTION
**Pull request by captainlee1024**
feat: adddocker file for batch-prover service (#94 )

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -5,5 +5,35 @@ services:
...
+    ports:
+      - "50051:50051"
+  scheduler-batch-prover-e2e-test:
+    build:
+      context: .
+      dockerfile: batch-prover.Dockerfile
+    volumes:
+      - ${HOST_PATH}:/./data
+    environment:
+      - SCHEDULER_ADDR=http://your-scheduler-address:50061
```
Comment (eigmax) [inline docker-compose.yml:37]: change your-scheduler-address to an env variable.

File: docker-compose.yml [hunk 2]
```diff
@@ -5,5 +5,55 @@ services:
...
+    build:
+      context: .
+      dockerfile: batch-prover.Dockerfile
+    volumes:
+      - ${HOST_DATA_PATH}:/app/data
+      - ${HOST_WORK_BASE}:/./starkjs
+    environment:
+      - SCHEDULER_ADDR=http://127.0.0.1:50051
+      - CONTINER_WORK_BASE=/.starkjs
+      - FORCE_BIT=21
```
Comment (eigmax) [inline docker-compose.yml:49]: use the latest force bit!

File: scheduler-test.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+RUN apt-get update && apt-get install -y \
+    autoconf \
+    automake \
+    libtool \
+    curl \
+    make \
+    gcc \
+    g++ \
+    unzip \
```
Comment (eigmax) [inline scheduler-test.Dockerfile:12]: remove unzip

Comment (eigmax) [reply scheduler-test.Dockerfile:12]: remove the curl

File: docker-compose.yml [hunk 4]
```diff
@@ -5,5 +5,55 @@ services:
...
+      - ${HOST_PATH}:/./data
+    environment:
+      - SCHEDULER_ADDR=http://127.0.0.1:50061
+  scheduler-test:
+    build:
+      context: .
+      dockerfile: scheduler-test.Dockerfile
+    volumes:
+      - ${HOST_DATA_PATH}:/app/data
+      - ${HOST_WORK_BASE}:/./starkjs
```
Comment (eigmax) [inline docker-compose.yml:26]: You need download the starkjs and [stark-circuits](https://github.com/0xEigenLabs/eigen-zkvm/tree/main/stark-circuits) from the scratch, not mapping the one from your computer!

File: scheduler-test.Dockerfile (hunk 3 above)
Comment (captainlee1024) [reply scheduler-test.Dockerfile]: done

File: docker-compose.yml (hunk 2 above)
Comment (captainlee1024) [reply docker-compose.yml]: done

File: batch-prover.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+RUN apt-get update && apt-get install -y \
+    autoconf \
+    automake \
+    libtool \
+    curl \
+    make \
+    gcc \
+    g++ \
+    unzip \
```
[the remark this answers is not in the document]
Comment (captainlee1024) [reply batch-prover.Dockerfile]: done

File: docker-compose.yml (hunk 4 above)
Comment (captainlee1024) [reply docker-compose.yml]: ok

**Review by eigmax — APPROVED** [review]

**Pull request merged**